### PR TITLE
Shared Events Creation Feature

### PR DESCRIPTION
**Pull request by fiyinfoluwaafol**
### Description
This PR consists of the feature that allows users to share events with each other within the app, such that it reflects on the invitees calendar. Invitees also have the option to view the events that have been shared with them, so that they can accept/decline shared events.

As for minor bugs encountered during development, because of how I implemented this feature it uses the same API endpoint for individual events, so I had to make sure that it didn't break the normal event creation functionality. In terms of edge cases, I made sure to account for proper error handling as it relates to the user inviting another user that doesn't exist, as well as to ensure that it doesn't create duplicate events if the user should include their email in the list of participants.

The next PR should include the implementation of editing shared events, such that it reflects on all instances of the events (such that it updates in the invitees' calendar)

### Milestones
This works towards the following user story:
User can share events

### Test Plan
Here is the video showing this PR's test plan (including all discussed features):
[Shared Events Creation Feature Implementation](https://www.loom.com/share/315d6b725270486ebbf93d924369f422?sid=79e5db69-f683-41ee-bc75-b3af3c7cad65)

### CODE REVIEW

File: backend/routes/calendarMgmt.js [hunk 1]
```diff
@@ -16,9 +16,18 @@ const upload = multer({ dest: 'uploads/' });
...
+    const {title, startAt, endAt, description, location, allDay, invitees} = req.body;
     try {
-    const newEvent = await prisma.calendarEvent.create({
+        // Retrieve the organizer's email
+        const organizer = await prisma.user.findUnique({
+            where: { id: parseInt(user_id) },
+            select: { email: true }
+        });
+        // Filter out the organizer's email from invitees if present
+        const filteredInvitees = invitees.filter(email => email !== organizer.email);
```
Comment (jameskim08) [inline backend/routes/calendarMgmt.js:27]: you can also gate this at the front end, by not allowing the user to enter their own email in the form ui.

File: frontend/src/components/SharedEventsPage/SharedEventsPage.jsx [hunk 2]
```diff
@@ -1,10 +1,31 @@
 import "./SharedEventsPage.css"
+import { useState, useEffect } from "react";
+import { fetchInvitations, respondToInvitation } from "../../utils/utils";
 
 function SharedEventsPage () {
+    const [invitations, setInvitations] = useState([]);
+
+    useEffect(() => {
+        fetchInvitations(setInvitations);
```
Comment (jameskim08) [inline frontend/src/components/SharedEventsPage/SharedEventsPage.jsx:9]: these fetch functions are good places to have loading states

File: frontend/src/components/SharedEventsPage/SharedEventsPage.jsx [hunk 3]
```diff
@@ -1,10 +1,31 @@
...
+    }, []);
+
+    const handleInvitationResponse = (eventId, status) => {
+        respondToInvitation(eventId, status);
+        setInvitations(invitations.filter(invite => invite.id !== eventId));
```
Comment (jameskim08) [inline frontend/src/components/SharedEventsPage/SharedEventsPage.jsx:14]: updating the database is also good places to have loading states

File: frontend/src/components/SharedEventsPage/SharedEventsPage.jsx (hunk 2 above)
Comment (jameskim08) [reply frontend/src/components/SharedEventsPage/SharedEventsPage.jsx:9]: should the fetchInvitations have a try/catch block on it? You should have some kind of error resolution here for the UI.

File: backend/routes/calendarMgmt.js (hunk 1 above)
Comment (juancafe4) [reply backend/routes/calendarMgmt.js:27]: it is a good practice to do in both: client and server

File: backend/routes/calendarMgmt.js [hunk 4]
```diff
@@ -80,6 +80,25 @@ router.post('/:user_id/events', authenticateToken, async (req, res) => {
...
 
+// API Endpoint for Fetching Event Invitation for Current User
+router.get('/invitations', authenticateToken, async (req, res) => {
+    const userId = req.user.id;
+
+    try {
+        const invitations = await prisma.calendarEvent.findMany({
+            where: {
+                userId,
+                status: 'pending',
```
Comment (juancafe4) [inline backend/routes/calendarMgmt.js:101]: you could create an "enum" for the status
```
 export const EventStatus = Object.freeze({
     ACCEPT: 'accept',
     PENDING: 'pending'
     DECLINE: 'decline'
});
```
or module.export.EventStatus

not a priority but an enum could avoid issues e.g somebody misspelled the event status and you are trying to find the issue. With an enum, it is only declared once

**Pull request merged**